sys/include/net/sock/tls.h

```diff
@@ -0,0 +1,119 @@
+/**
+ * @file tls.h
+ * @brief TLS over TCP socket implementation using wolfSSL for RIOT OS
+ * @author Bilal-Alali
+ * @date 2025-05-21 12:35:22
+ */
+
+#ifndef NET_SOCK_TLS_H
+#define NET_SOCK_TLS_H
+
+#include "net/sock/tcp.h"
+#include <wolfssl/ssl.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief TLS Socket structure encapsulating TCP and wolfSSL components
+ */
+typedef struct sock_tls_tcp {
+    sock_tcp_t *tcp_sock;    /**< Underlying TCP socket for network communication */
+    WOLFSSL_CTX *ctx;        /**< WolfSSL context for TLS configuration */
+    WOLFSSL *ssl;            /**< WolfSSL session for active TLS connection */
+} sock_tls_tcp_t;
+
+/**
+ * @brief TLS Socket queue structure for managing incoming connections
+ */
+typedef struct sock_tls_tcp_queue {
+    sock_tcp_queue_t tcp_queue;  /**< Underlying TCP queue for connection handling */
+    WOLFSSL_CTX *ctx;           /**< WolfSSL context shared across connections */
+} sock_tls_tcp_queue_t;
```

Comment:
Note: This adds implementation details to the common header. It would be better to move WolfSSL specifics somewhere into `pkg/wolfssl`.

The idea of the SOCK APIs is to have a software stack independent interface, so that applications could easily switch between stacks. E.g. using `sock_udp` would work the same way for GNRC as it does for lwIP; it just requires selecting different modules. IMO splitting out the types into a separate header and included that from this one would make life easier for anyone who wanted to add a 2nd implementation of TLS.